test/tests/p5.Metro.js

```diff
@@ -0,0 +1,85 @@
+const expect = chai.expect;
+let metro;
+
+describe('p5.Metro', function () {
+  beforeEach(function () {
+    metro = new p5.Metro();
+  });
+
+  it('can be created', function () {
+    expect(metro).to.have.property('bpm').to.equal(120);
+    expect(metro).to.have.property('clock');
+    expect(metro).to.have.property('syncedParts').to.be.an('array');
+  });
+
+  it('can be initialised with a beatlength, bpm', function () {
+    metro.beatLength(0.0625);
+    metro.setBPM(60);
+    expect(metro.tatums).to.equal(4);
+    expect(metro.tatumTime).to.equal(0.25);
+    expect(metro.getBPM()).to.equal(60);
+  });
+
+  it('can be started and stopped', function (done) {
+    this.timeout = 2000;
+    let ticks;
+    metro.setBPM(600);
+    metro.start();
+    setTimeout(() => {
+      ticks = metro.metroTicks;
+      metro.stop();
+      expect(ticks).to.not.equal(0);
+      setTimeout(() => {
+        expect(metro.metroTicks).to.equal(ticks);
+        done();
+      }, 100);
+    }, 1000);
+  });
+
+  it('can be started and stopped with dealy', function (done) {
+    let ticks;
+    metro.setBPM(600);
+    metro.start(0.1);
+    setTimeout(() => {
+      ticks = metro.metroTicks;
+      metro.stop(0.2);
+      expect(ticks).to.not.equal(0);
+      setTimeout(() => {
+        expect(metro.metroTicks).to.not.equal(ticks);
```

Comment:
```suggestion
        expect(metro.metroTicks).to.be.above(ticks);
```

In an ideal world we would test that the value is equal to an exact value. But since we are using the JS clock's setTimeout, which is not in sync or as precise as the AudioContext time, we can't really know what that value will be, but "greater than" feels more specific than "not equal" which could mean lower or higher.

Reply:
Agreed